src/blocks/blocks/stripe-checkout/edit.tsx

```diff
@@ -128,51 +164,115 @@ const Edit = ({
 		});
 	}, [ products, prices, attributes.price ]);
 
-	const showPlaceholder = ( isLoadingProducts || isLoadingPrices || hasProductsRequestFailed || hasPricesRequestFailed || undefined === attributes.product || undefined === attributes.price );
+	const showPlaceholder = ( isLoadingProducts || isLoadingPrices || hasProductsRequestFailed || hasPricesRequestFailed || undefined === attributes.product || undefined === attributes.price || 'loaded' !== status || ! hasValidApiKey() );
 
 	const blockProps = useBlockProps({
 		className: classnames({ 'is-placeholder': showPlaceholder })
 	});
 
+	const [ apiKey, setAPIKey ] = useState( '' );
+
+	const saveApiKey = () => {
+		updateOption( 'themeisle_stripe_api_key', apiKey, __( 'Stripe API Key saved!', 'otter-blocks' ), async() => {
+			dispatch( 'themeisle-gutenberg/data' ).invalidateResolutionForStoreSelector( 'getStripeProducts' );
+			dispatch( 'themeisle-gutenberg/data' ).invalidateResolutionForStoreSelector( 'getStripeProductPrices' );
+			setAPIKey( '' );
+		});
+	};
+
 	if ( showPlaceholder ) {
 		return (
 			<div { ...blockProps }>
 				<Placeholder
 					icon={ store }
 					label={ __( 'Stripe Checkout', 'otter-blocks' ) }
-					instructions={ ( hasProductsRequestFailed && productsError?.message ) || ( hasPricesRequestFailed && pricesError?.message ) }
+					instructions={ ( hasProductsRequestFailed && productsError?.message ) || ( hasPricesRequestFailed && pricesError?.message ) || ( hasPricesRequestFailed && pricesError?.message ) }
```

Comment:
It can be a small notice as currently the text is really ignorable and doesn't alert you at all, took me time to notice it appearing.